documentation/src/main/asciidoc/reference/mapper-orm-indexing-massindexer.asciidoc

```diff
@@ -123,6 +123,32 @@ make sure you size your connection pool according to the mass indexing parameter
 as explained in <<mapper-orm-indexing-massindexer-tuning-threads>>.
 ====
 
+[[mapper-orm-indexing-massindexer-conditional]]
+== Conditional reindexing
+You can also select a subset of target entities to be reindexed.
+
+.Use of Conditional reindexing
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/indexing/HibernateOrmMassIndexerIT.java[tags=reindexOnly]
+----
+<1> Get the `SearchSession`.
+<2> Create a `MassIndexer` targeting every indexed entity type.
+<3> Reindex only the books published before year 2100.
+<4> Reindex only the authors born prior to a given local date.
+<5> In this example the date is passed as a query parameter.
+<6> Start the mass indexing process and return when it is over.
+====
+
+[WARNING]
+====
+Even if the reindexing is applied on a subset of entities, by default *all entities* will be purged at the start.
+At the moment there is no way to filter the entities that will be purged.
```

Comment:
I just noticed this could lead people to think entities are always purged; maybe explain that purge can be disabled?

```suggestion
The purge <<mapper-orm-indexing-massindexer-parameters,can be disabled completely>>, but when enabled there is no way to filter the entities that will be purged.
```